### ISSUE TITLE
signal: Implement sigaction

### ISSUE THREAD

**Issue by dlrobertson** [redox-os/relibc#101]
`sigaction` is not currently implemented, and is needed for implementing `signal` etc.

- [ ] Implement basic wrappers for the `sigaction` syscall for linux and redox in the `platform` crate.
- [ ] Implement `restorer` for linux. (It looks like [redox-os/syscall] already implements `sigrestore`) 
- [ ] Implement `sigaction` in the `signal` crate.

[redox-os/syscall]: https://github.com/redox-os/syscall

**Comment by dlrobertson** [comment]
**Note:** This issue may not be a great first issue. It might require writing some x86_64 and aarch64 assembly to implement the linux signal restorer.

**Comment by Tommoa** [comment]
According to http://man7.org/linux/man-pages/man2/sigaction.2.html the `sa_restorer` function pointer is a pointer to `sigreturn()` (http://man7.org/linux/man-pages/man2/sigreturn.2.html). 

From further looking into the musl source, the x86_64 version just calls the `rt_sigreturn` syscall (https://git.musl-libc.org/cgit/musl/tree/src/signal/x86_64/restore.s), and while I am not as well versed in aarch assembly, the comments lead me to believe that it does the same thing as well. Couldn't we just have a function while simply calls the syscall versions of `sigreturn` (https://docs.rs/redox_syscall/0.1.37/syscall/call/fn.sigreturn.html in redox and https://docs.rs/sc/0.2.2/sc/platform/nr/constant.RT_SIGRETURN.html in linux) and have our sigaction always set the `SA_RESTORER` flag (as musl does)?

Is there something I'm missing that takes this from relatively trivial to a substantially more complex problem?

**Comment by dlrobertson** [comment]
Good point! Your analysis seems correct. aarch64 does the same thing https://git.musl-libc.org/cgit/musl/tree/src/signal/aarch64/restore.s

**Comment by Tommoa** [comment]
@dlrobertson want me to do it?

**Comment by dlrobertson** [comment]
Sure! That would be awesome!
